llvm/include/llvm/Analysis/TargetLibraryInfo.h

```diff
@@ -565,6 +567,16 @@ class TargetLibraryInfo {
   /// \copydoc TargetLibraryInfoImpl::getSizeTSize()
   unsigned getSizeTSize(const Module &M) const { return Impl->getSizeTSize(M); }
 
+  /// Returns an IntegerType corresponding to size_t.
+  IntegerType *getSizeTType(const Module &M) const {
+    return IntegerType::get(M.getContext(), getSizeTSize(M));
+  }
+
+  /// Returns a constant materialized as a size_t type.
+  ConstantInt *getAsSizeT(uint64_t V, const Module &M) const {
```

Comment:
(suggestion) Maybe just `getSizeT`? It would be more in line with IRBuilder methods.

Reply:
I avoided that as I felt the name was ambiguous as to whether it returned a value or a type.